src/explorer.ts

```diff
@@ -0,0 +1,130 @@
+import * as extensionApi from '@podman-desktop/api';
+import { mkdtemp, readFile, rm } from 'node:fs/promises';
+import * as path from 'node:path';
+import * as os from 'node:os';
+import * as nodeTar from 'tar';
+
+const WHITEOUT_MARKER = '.wh.';
+const OPAQUE_WHITEOUT_MARKER = '.wh..wh..opq';
+
+interface History {
+  created_by?: string;
+  empty_layer?: boolean;
+}
+
+export class Explorer {
+
+  #provider: extensionApi.ImageFilesProvider | undefined;
+
+  setProvider(provider: extensionApi.ImageFilesProvider) {
+    this.#provider = provider;
+  }
```

Comment:
should it be a constructor thingy ? rather than a setter

Reply:
I would prefer, but I don't think it is possible, as we have to build the Explorer before we have the `provider` value:

```
const explorer = new Explorer();
provider = extensionApi.provider.createImageFilesProvider({
  getFilesystemLayers: explorer.getFilesystemLayers.bind(explorer),
});
explorer.setProvider(provider);
```